airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/manifest.yaml

```diff
@@ -0,0 +1,311 @@
+      path: "account"
+  # SHORTCUTS
+  shortcuts_stream:
+    description: >-
+      Shortcuts Stream: https://developer.zendesk.com/rest_api/docs/chat/shortcuts#list-shortcuts
+    $ref: "#/definitions/base_stream_with_list_response_no_pagination"
+    $parameters:
+      name: "shortcuts"
+      path: "shortcuts"
+  # ROUTING SETTINGS
+  routing_settings_stream:
+    description: >-
+      Routing Settings Stream: https://developer.zendesk.com/rest_api/docs/chat/routing_settings#show-account-routing-settings
+    $ref: "#/definitions/base_stream"
+    primary_key: ""
+    $parameters:
+      name: "routing_settings"
+      data_field: "data"
+      path: "routing_settings/account"
+  # TRIGGERS
+  triggers_stream:
+    description: >-
+      Triggers Stream: https://developer.zendesk.com/rest_api/docs/chat/triggers#list-triggers
+    $ref: "#/definitions/base_stream_with_list_response_no_pagination"
+    $parameters:
+      name: "triggers"
+      path: "triggers"
+  # ROLES
+  roles_stream:
+    description: >-
+      Roles Stream: https://developer.zendesk.com/rest_api/docs/chat/roles#list-roles
+    $ref: "#/definitions/base_stream_with_list_response_no_pagination"
+    $parameters:
+      name: "roles"
+      path: "roles"
+  # SKILLS
+  skills_stream:
+    description: >-
+      Skills Stream: https://developer.zendesk.com/rest_api/docs/chat/skills#list-skills
+    $ref: "#/definitions/base_stream_with_list_response_no_pagination"
+    $parameters:
+      name: "skills"
+      path: "skills"
+  # GOALS
+  goals_stream:
+    description: >-
+      Goals Stream: https://developer.zendesk.com/rest_api/docs/chat/goals#list-goals
+    $ref: "#/definitions/base_stream_with_list_response_no_pagination"
+    $parameters:
+      name: "goals"
+      path: "goals"
+  # DEPARTMENTS
+  departments_stream:
+    description: >-
+      Departments Stream: https://developer.zendesk.com/rest_api/docs/chat/departments#list-departments
+    $ref: "#/definitions/base_stream_with_list_response_no_pagination"
+    $parameters:
+      name: "departments"
+      path: "departments"
+
+  # INCREMENTAL STREAMS
+  # AGENTS
+  agents_stream:
+    description: >-
+      Agents Stream: https://developer.zendesk.com/rest_api/docs/chat/agents#list-agents
+    $ref: "#/definitions/base_incremental_id_stream"
+    $parameters:
+      name: "agents"
+      path: "agents"
+  # BANS
+  bans_stream:
+    description: >-
+      Bans Stream: https://developer.zendesk.com/rest_api/docs/chat/bans#list-bans
+    $ref: "#/definitions/base_incremental_id_stream"
+    retriever:
+      $ref: "#/definitions/base_incremental_id_stream/retriever"
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: CustomRecordExtractor
+          class_name: source_zendesk_chat.components.bans_record_extractor.ZendeskChatBansRecordExtractor
+    $parameters:
+      name: "bans"
+      path: "bans"
+  # AGENTS TIMELINES
+  agents_timelines_stream:
```

Comment:
Differences observed:
* Query param `start_time` changes from `1601510400000000` to `1601524800000000` (this will probably give you another value depending on your timezone). So I've learned today that parsing to `%s` format ignores the timezone so that `datetime.datetime.fromisoformat("2020-10-01T00:00:00+00:00").strftime("%s")` will return an epoch assuming `2020-10-01T00:00:00` was in your timezone and the `+00:00`. Based on the [IBM documentation](https://www.ibm.com/docs/en/i/7.3?topic=functions-strftime-convert-datetime-string#strfti__title__5), this behavior can be affected by the system timezone. So I guess there is a bug in the CDK which we should fix by adding [this if](https://github.com/airbytehq/airbyte/blob/0113c525548080f128258042d8663a415897ef8c/airbyte-cdk/python/airbyte_cdk/sources/declarative/datetime/datetime_parser.py#L28-L29) [here](https://github.com/airbytehq/airbyte/blob/0113c525548080f128258042d8663a415897ef8c/airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py#L108-L117). I have created [an issue](https://github.com/airbytehq/airbyte-internal-issues/issues/6545) for that
* The id changes from `360786799676|2020-12-14T04:08:32Z` to `360786799676|2020-12-14T04:08:32.301292Z`. I'm not sure why as [the code from the previous version](https://github.com/airbytehq/airbyte/blob/0113c525548080f128258042d8663a415897ef8c/airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/streams.py#L210) seems very similar. Can you investigate on that one?

Reply:
Given my understanding, it could lead to duplication of records if we do not fix this and don't ask the user to reset the stream. The case would be:
* First sync on non low-code version, state cursor value is the one of the last record
* Source is updated
* Sync a second time from the last record (so last record will be picked up again) but ID created is different from the first sync so dedup is not happening

Can you try implementing the fix? I'm up-to-date on `origin/baz/source/zendesk-chat/migrate-to-low-code` and still see the issue. Or we ask the user to reset?

Reply:
It's not the `ID` has changed, the actual change is in the `start_time`, but because the `id` was missing for this stream, as a PK, so i've added it to the schema. It was not my change or something, i've just transferred the logic from the old code base to the low-code.

Reply:
Here are the methods from Python Version of the `Source Zendesk Chat`:
- [Convertion to different format](https://github.com/airbytehq/airbyte/blob/1d6b93d6346de38f140fc1ef6b1a736902f4941c/airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/streams.py#L100-L107)
- [applying STATE level `cursor_field` format mutation](https://github.com/airbytehq/airbyte/blob/1d6b93d6346de38f140fc1ef6b1a736902f4941c/airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/streams.py#L123-L128)
- [applying RECORD level `cursor_field` format mutation](https://github.com/airbytehq/airbyte/blob/1d6b93d6346de38f140fc1ef6b1a736902f4941c/airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/streams.py#L149-L151)

These convert the date-time with microseconds to the RFC-3339 without microseconds.

1) Should I create another Custom Components to cover this and apply this as a separate record_extractor and make the streams to produce the: `"2020-12-14T04:08:32Z"` instead of `"2020-12-14T04:08:32.301292Z"`
2) Should I make the same with the STATE?

Reply:
Ok! So we basically lose precision because of how we parse datetime, right? I honestly don't mind keeping the behavior of the low-code implementation. This is a breaking change though. I don't know what is best: we do it as part of this PR or in another issue? I'll let you decide what is best